tidy3d/components/medium.py

```diff
@@ -667,6 +667,48 @@ def time_modulated(self) -> bool:
         """Whether any component of the medium is time modulated."""
         return self.modulation_spec is not None and self.modulation_spec.applied_modulation
 
+    @cached_property
+    def nonlinear(self) -> bool:
```

Comment:
I'd prefer `is_nonlinear`, `is_custom`, etc. Minor point bakes it more clear these are `bool` and also leaves these names available in case we need them later.

Reply:
I might leave time_modulated instead of changing it to is_time_modulated, just to avoid the need for a backend PR.

Reply:
hm, might be confusing if they follow different conventions?